src/ml/neural_net/image_augmentation.cpp

```diff
@@ -117,5 +117,98 @@ image_augmenter::result resize_only_image_augmenter::prepare_images(
   return res;
 }
 
+image_augmenter::result processed_image_augmenter::prepare_images(
+    std::vector<labeled_image> source_batch) {
+  result res;
+
+  const size_t n = opts_.batch_size;
+  constexpr size_t c = 3;
+
+  std::vector<turi::neural_net::shared_float_array> images_to_aug;
+  std::vector<turi::neural_net::shared_float_array> ann_batch;
+  std::vector<turi::neural_net::shared_float_array> pred_batch;
+
+  // Decode a batch of images to raw format
+  for (size_t i = 0; i < n; i++) {
+    size_t input_height = source_batch[i].image.m_height;
```

Comment:
Please factor out all the instances of `source_batch[i]`. There's no need to perform a new array lookup each time